arbitrum-docs/for-devs/quickstart-solidity-hardhat.md

```diff
@@ -0,0 +1,491 @@
+
+<VendingMachine id='dumb-cupcakes' type='web2' />
+
+Note that although this vending machine appears to follow the rules, it doesn't follow them as much as we'd like. The vending machine's business logic and data are hosted by a **centralized service provider**. We're trusting that this service provider isn't malicious, but:
+
+ 1. Our centralized service provider can deny access to particular users.
+ 2. A malicious actor can change the rules of the vending machine at any time, for example, to give their friends extra cupcakes.
+  
+Centralized third-party intermediaries represent a **single point of failure** that malicious actors may become incentivized to exploit. To mitigate this type of risk, we can decentralize our vending machine's **business logic and data**, rendering this type of exploitation infeasible.
+
+This is Arbitrum's core value proposition to you, dear developer. Arbitrum makes it easy for you to deploy your vending machines to Ethereum's permissionless, <a data-quicklook-from='trustless'>trustless</a>, *decentralized* network of nodes[^2] while keeping costs low for you and your users.
+
+Let's implement the "web3" version of the above vending machine using Arbitrum.
+
+### Prerequisites
+
+- **VS Code**: The IDE we'll use to build our vending machine. See [code.visualstudio.com](https://code.visualstudio.com/) to install.
+- **Metamask**: The <a data-quicklook-from='ethereum-wallet'>wallet</a> we'll use to interact with our vending machine. See [metamask.io](https://metamask.io/) to install.
+- **Yarn**: The package manager we'll use to install our dependencies. See [yarnpkg.com](https://yarnpkg.com/) to install.
+
+We'll install the rest of our dependencies as we go.
+
+### Ethereum and Arbitrum in a nutshell
+
+ - Ethereum is a decentralized network of <a href='https://docs.prylabs.network/docs/concepts/nodes-networks'>nodes</a> that use Ethereum's client software (like Offchain's <a href='https://docs.prylabs.network/docs/getting-started'>Prysm</a>) to maintain a public <a data-quicklook-from='blockchain'>blockchain</a> data structure.
+ - The data within Ethereum's blockchain data structure changes one transaction at a time.
+ - <a data-quicklook-from='smart-contract'>Smart contracts</a> are small programs that execute transactions according to predefined rules. Ethereum's nodes host and execute smart contracts.
+ - You can use smart contracts to build decentralized apps (dApps) that use Ethereum's network to process transactions and store data.
+ - DApps let users carry their data and identity between applications without having to trust centralized service providers.
+ - People who run Ethereum nodes[^3] can receive rewards for processing and validating transactions on behalf of users and dApps.
+ - These transactions can be expensive when the network is under heavy load. **Layer 2** (L2) scaling solutions like Arbitrum solve this problem.
```

Comment:
Should we probably specify that Arbitrum is an _optimistic rollup_ among different types of the L2 scaling solutions?